frontend/public/components/utils/details-page.tsx

```diff
@@ -23,7 +24,7 @@ const getTolerations = (obj: K8sResourceKind): Toleration[] => {
     : _.get(obj, 'spec.template.spec.tolerations');
 };
 
-export const ResourceSummary: React.SFC<ResourceSummaryProps> = ({children, resource, showPodSelector = false, showNodeSelector = false, showAnnotations = true, showTolerations = false, podSelector = 'spec.selector'}) => {
+export const ResourceSummary: React.SFC<ResourceSummaryProps> = ({children, id, resource, showPodSelector = false, showNodeSelector = false, showAnnotations = true, showTolerations = false, podSelector = 'spec.selector'}) => {
```

Comment:
Why do we need a different id/prefix for each `ResourceSummary`? Do we ever have more than one on a page?

I'd like use values like `data-test-id="resource-summary-name"`, etc., and skip the prefixes. The prefixes will cause problems in generic tests like crud scenario that test many different types.

Reply:
> Why do we need a different id/prefix for each `ResourceSummary`? Do we ever have more than one on a page?
> 

No, but it is written this way so we could do that in the future.

> I'd like use values like `data-test-id="resource-summary-name"`, etc., and skip the prefixes. The prefixes will cause problems in generic tests like crud scenario that test many different types.

The generic tests could still take that into account. The `data-test-id` approach is only usable in special cases like this one. 
Many components expect only `id` attribute (which is also tested) plus it is not a good practice to pass special test data down the component tree. Using just IDs would make testing more consistent.

Reply:
> No, but it is written this way so we could do that in the future.

Let's add that only when [we need it](http://wiki.c2.com/?YouArentGonnaNeedIt).

> The generic tests could still take that into account. The data-test-id approach is only usable in special cases like this one.

I don't really understand the complaint here. If this is only used for tests, we should be make that clear. `id` is frequently used for other behavioral things (label for, CSS styles, referenced in JS code, etc).

> Many components expect only id attribute (which is also tested) plus it is not a good practice to pass special test data down the component tree. 

We're not passing anything down the component tree. We just add the data-test-id we need directly in the component.

> Using just IDs would make testing more consistent.

Why? It's actually inconsistent since we already use data-test-id in tests today.

Reply:
The other problem with `id` is that it has to be unique on the page. `data-test-id` does not, so you can use the component more than once (and still find the one you need using a more precise selector).

Reply:
> > No, but it is written this way so we could do that in the future.
> 
> Let's add that only when [we need it](http://wiki.c2.com/?YouArentGonnaNeedIt).
> 

This was not a big feature for the future, just an attempt to make the component wholesome.

> > The generic tests could still take that into account. The data-test-id approach is only usable in special cases like this one.
> 
> I don't really understand the complaint here. If this is only used for tests, we should be make that clear. `id` is frequently used for other behavioral things (label for, CSS styles, referenced in JS code, etc).
> 
> > Many components expect only id attribute (which is also tested) plus it is not a good practice to pass special test data down the component tree.
> 
> We're not passing anything down the component tree. We just add the data-test-id we need directly in the component.

Ik, I was just talking about what would happen if we used only this approach.
> 
> > Using just IDs would make testing more consistent.
> 
> Why? It's actually inconsistent since we already use data-test-id in tests today.

You are right, let's use both approaches then. Closing this PR.

> The other problem with id is that it has to be unique on the page. data-test-id does not, so you can use the component more than once (and still find the one you need using a more precise selector).

This PR took that into account.